### PR TITLE
fix(workflow): Remove tracked superuser views

### PR DESCRIPTION
**Pull request by iProgramStuff**
We recently added code to only track alert views of people in the org + member of associated alert projects.

This migration removes all of the views we've tracked before adding this change (i.e. superuser views of orgs we're not a part of).

### CODE REVIEW

File: src/sentry/migrations/0067_remove_tracked_superusers.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        tracked_incident = tracked_view.incident
+        is_org_member = tracked_incident.organization.has_access(tracked_user)
+        is_project_member = False
+        if is_org_member:
+            for incident_project in IncidentProject.objects.filter(incident=tracked_incident).select_related(
+                "project"
+            ):
+                if incident_project.project.member_set.filter(user=tracked_user).exists():
+                    is_project_member = True
+                    break
```
Comment (wedamija) [inline src/sentry/migrations/0067_remove_tracked_superusers.py:26]: I think we should probably skip this check. There are manager/owner level permissions in an org, and they're allowed to see things related to projects they're not explicit members of. `organization.has_access` checks that they're a member of the org, which should be good enough to remove only superusers.

File: src/sentry/migrations/0067_remove_tracked_superusers.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from django.db import migrations
+
+def remove_tracked_superuser_views(apps, schema_editor):
+    """
+    We recently added code to only track alert views of people in the org + member of associated alert projects.
+    This migration removes all of the views we've tracked before adding this change (i.e. superuser views of orgs we're not a part of).
+    """
+    IncidentSeen = apps.get_model("sentry", "IncidentSeen")
+    IncidentProject = apps.get_model("sentry", "IncidentProject")
+    tracked_views = IncidentSeen.objects.all()
```
Comment (markstory) [inline src/sentry/migrations/0067_remove_tracked_superusers.py:14]: How many records would this pull up? Do you have a risk of running out of memory?

File: src/sentry/migrations/0067_remove_tracked_superusers.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    for tracked_view in tracked_views:
+        tracked_user = tracked_view.user
+        tracked_incident = tracked_view.incident
```
Comment (markstory) [inline src/sentry/migrations/0067_remove_tracked_superusers.py:17]: These will emit additional queries. You might want to eager load these associations.

File: src/sentry/migrations/0067_remove_tracked_superusers.py (hunk 2 above)
Comment (iProgramStuff) [reply src/sentry/migrations/0067_remove_tracked_superusers.py:14]: I'm thinking no. There are 384 records in this case...but for the future, I'll try to keep this in mind. What kind of memory limits do we actually have? How many records do you think is a reasonable amount to hold in memory?

Comment (wedamija) [reply src/sentry/migrations/0067_remove_tracked_superusers.py:14]: Typically we'd use `RangeQuerySetWrapperWithProgressBar` to limit the number of items in memory. The default pagination size for that is 1000, so probably once we get above that.

File: src/sentry/migrations/0069_remove_tracked_superusers.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from django.db import migrations
+
+def remove_tracked_superuser_views(apps, schema_editor):
+    """
+    We recently added code to only track alert views of people in the org + member of associated alert projects.
+    This migration removes all of the views we've tracked before adding this change (i.e. superuser views of orgs we're not a part of).
+    """
+    IncidentSeen = apps.get_model("sentry", "IncidentSeen")
+    IncidentProject = apps.get_model("sentry", "IncidentProject")
+    tracked_views = IncidentSeen.objects.all().select_related("user").select_related("incident")
```
Comment (wedamija) [inline src/sentry/migrations/0069_remove_tracked_superusers.py:14]: Nit: You can list multiple tables in the same `select_related`, call, like `select_related("user", "incident", ...)`

File: src/sentry/migrations/0069_remove_tracked_superusers.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    atomic = True
```
Comment (wedamija) [inline src/sentry/migrations/0069_remove_tracked_superusers.py:39]: Let's set this to `False`

**Review by markstory — APPROVED** [review]

**Pull request merged**